Deprecated since NumPy 1.25: conversion of an array with ndim > 0 to a scalar

```python
/usr/local/lib/python3.10/site-packages/ewoksxrpd/tasks/calibrate.py:326: in fit
    setup.refine3(fix=fixed)  # method="slsqp"
/usr/local/lib/python3.10/site-packages/pyFAI/goniometer.py:970: in refine3
    old_delta_theta2 = self.residu3(param, free, const)
/usr/local/lib/python3.10/site-packages/pyFAI/goniometer.py:864: in residu3
    pyFAI_param.append(single_param.get("wavelength", self.wavelength) * 1e10)
/usr/local/lib/python3.10/site-packages/pyFAI/goniometer.py:1042: in get_wavelength
    return Goniometer.get_wavelength(self)
/usr/local/lib/python3.10/site-packages/pyFAI/goniometer.py:447: in get_wavelength
    return self.trans_function(self.param, dummy_position).wavelength
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = ExtendedTransformation with param: ('dist_offset', 'dist_scale', 'poni1_offset', 'poni1_scale', 'poni2_offset', 'poni2...* x + poni2_offset / 100
    rot1= arot1 / 10
    rot2= arot2 / 10
    rot3= arot3 / 10
    wavelength= hc*1e-10/energy
param = [0.0, 1.0100000000000002, 10.086, 0.0, 9.914000000000001, 0.0, ...]
pos = [0]
    def __call__(self, param, pos):
        """This makes the class instance behave like a function,
        actually a function that translates the n-parameter of the detector
        positioning on the goniometer and the m-parameters.
    
        :param param: parameter of the fit
        :param pos: position of the goniometer (representation from the
            goniometer)
        :return: 6-tuple with (dist, poni1, poni2, rot1, rot2, rot3) as needed
            for pyFAI.
        """
        res = {}
        variables = self.variables.copy()
        for name, value in zip(self.param_names, param):
            variables[name] = value
        if len(self.pos_names) == 1:
            variables[self.pos_names[0]] = pos
        else:
            for name, value in zip(self.pos_names, pos):
                variables[name] = value
        for name, code in self.codes.items():
            signa = [variables.get(name, numpy.NaN) for name in code.input_names]
>           res[name] = (float(code(*signa)))
E           DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
```

--- a/src/pyFAI/goniometer.py
+++ b/src/pyFAI/goniometer.py
@@ -34,7 +34,7 @@ __author__ = "Jérôme Kieffer"
 __contact__ = "Jerome.Kieffer@ESRF.eu"
 __license__ = "MIT"
 __copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
-__date__ = "20/02/2024"
+__date__ = "01/04/2024"
 __status__ = "development"
 __docformat__ = 'restructuredtext'
 
@@ -232,8 +232,8 @@ class GeometryTransformation(object):
                 variables[name] = value
         for name, code in self.codes.items():
             signa = [variables.get(name, numpy.nan) for name in code.input_names]
-            res[name] = (float(code(*signa)))
-            # could ne done in a single liner but harder to understand !
+            res[name] = (float(code(*signa)[()]))
+            # could be done in a single liner but harder to understand !
         return PoniParam(**res)
 
     def __repr__(self):
@@ -357,7 +357,7 @@ class ExtendedTransformation(object):
                 variables[name] = value
         for name, code in self.codes.items():
             signa = [variables.get(name, numpy.nan) for name in code.input_names]
-            res[name] = (float(code(*signa)))
+            res[name] = (float(code(*signa)[()]))
             # could ne done in a single liner but harder to understand !
         return self.ParamNT(**res)
 
@@ -690,7 +690,7 @@ class SingleGeometry(object):
                 break
             if count[i]:
                 rings += 1
-                mask = qmask==i
+                mask = qmask == i
                 sub_data = self.image[mask]
                 mean = sub_data.mean(dtype=numpy.float64)
                 std = sub_data.std(dtype=numpy.float64)
@@ -735,9 +735,11 @@ class SingleGeometry(object):
     def get_wavelength(self):
         assert self.calibrant.wavelength == self.geometry_refinement.wavelength
         return self.geometry_refinement.wavelength
+
     def set_wavelength(self, value):
         self.calibrant.setWavelength_change2th(value)
         self.geometry_refinement.set_wavelength(value)
+
     wavelength = property(get_wavelength, set_wavelength)
 
 
@@ -926,8 +928,6 @@ class GoniometerRefinement(Goniometer):
             print(self.nt_param(*self.param))
         return self.param
 
-
-
     def refine3(self, fix=None, method="slsqp", verbose=True, **options):
         """Geometry refinement tool
 
@@ -988,7 +988,6 @@ class GoniometerRefinement(Goniometer):
             return new_delta_theta2
         else:
             return old_delta_theta2
-
 
     def set_bounds(self, name, mini=None, maxi=None):
         """Redefines the bounds for the refinement

